### DISCUSSION TITLE
How many report types on cypress ?

### DISCUSSION THREAD

**Discussion by adnanerlansyah403** [cypress-io/cypress#28487] [Questions and Help]
Hello everyone, I want to ask about how many report types exist on Cypress. for example, as we know cypress can generate the reports immediately on our local machine by using the package cypress-mochawesome-report and we can use cypress cloud too. So my question is, what are the terms you'll call this report type?

**Comment by portyu9** [comment]
Hi @adnanerlansyah403 

These are better thought of as **two different reporting mechanisms**, rather than two formats of the same report.

### 1. Test reporters

Examples:

- Cypress's default `spec` reporter
- JUnit
- TeamCity
- Mochawesome
- other Mocha/custom reporters

For example:

```js
reporter: 'mochawesome'
```

These are usually called **test reporters**, **Mocha reporters**, or **custom reporters**. They produce console output or local/CI files such as JSON, XML, and HTML.

### 2. Cypress Cloud recorded runs

When you run Cypress with recording enabled, test results and supported artifacts are sent to **Cypress Cloud**.

That is normally called a **recorded Cypress Cloud run**, not a reporter.

Cypress Cloud provides hosted run history, test results, analytics, artifacts, parallelization/orchestration, etc.

So for your examples:

- `cypress-mochawesome-report` → a **test reporter / Mochawesome reporter**
- Cypress Cloud result page → a **recorded Cypress Cloud run**

They can also be used together: a CI run can generate a local Mochawesome report while also recording the run to Cypress Cloud.

Please mark this answer as accepted if it helped, thank you.
